lib/babe/epoch_handler.go

```diff
@@ -26,16 +27,15 @@ type epochHandler struct {
 	constants constants
 	epochData *epochData
 
-	// for slots where we are a producer, store the vrf output (bytes 0-32) + proof (bytes 32-96)
-	slotToProof map[uint64]*VrfOutputAndProof
+	slotToPreRuntimeDigest map[uint64]*types.PreRuntimeDigest
 
 	handleSlot handleSlotFunc
 }
 
 func newEpochHandler(epochNumber, firstSlot uint64, epochData *epochData, constants constants,
 	handleSlot handleSlotFunc, keypair *sr25519.Keypair) (*epochHandler, error) {
 	// determine which slots we'll be authoring in by pre-calculating VRF output
-	slotToProof := make(map[uint64]*VrfOutputAndProof, constants.epochLength)
+	slotToPreRuntimeDigest := make(map[uint64]*types.PreRuntimeDigest)
```

Comment:
keep the pre-allocation?

```suggestion
	slotToPreRuntimeDigest := make(map[uint64]*types.PreRuntimeDigest, constants.epochLength)
```